tiledb/sm/array_schema/attribute.cc

```diff
@@ -46,15 +46,10 @@ namespace sm {
 Attribute::Attribute() = default;
 
 Attribute::Attribute(const char* name, Datatype type) {
-  // Set name
   if (name != nullptr)
     name_ = name;
-
-  // Set type
   type_ = type;
-
-  // Set default compressor and compression level
-  cell_val_num_ = 1;
+  cell_val_num_ = datatype_is_string(type) ? constants::var_num : 1;
```

Comment:
Why is the variable-sized the default for strings? That differs from every other datatype (and I'm not sure I see the motivation for why).